crypto/rand/rand_lib.c

```diff
@@ -25,6 +25,45 @@ static CRYPTO_RWLOCK *rand_meth_lock;
 static const RAND_METHOD *default_RAND_meth;
 static CRYPTO_ONCE rand_init = CRYPTO_ONCE_STATIC_INIT;
 
+#ifdef OPENSSL_RAND_SEED_RDTSC
+/*
+ * Since we get some randomness from the low-order bits of the
+ * high-speec clock, it can help.  But don't return a status since
+ * it's not sufficient to indicate whether or not the seeding was
+ * done.
+ */
+void rand_rdtsc(void)
+{
+    unsigned char c;
+    int i;
+
+    for (i = 0; i < 10; i++) {
+        c = (unsigned char)(OPENSSL_rdtsc() & 0xFF);
```

Comment:
Yes, the SP800-90B process require an _a priori_ estimate of the randomness.  The statistics the tests and estimates are based on remain valid without such.  Given the collected samples pass more stringent tests and the low sampling rates used, 8 bits in the low order byte isn't an unreasonable expectation.  I'd be fairly confident about low 10 or 12 bits, although I didn't analyse that far.

All of the suites I know of, except the _ent_ program, pick up linear changes well.  Knuth's runs test would be expected to identify such.  Other tests would too I expect.

I've also some empirical evidence to match: low microsecond sampling intervals of TSC failed everything miserably.